### PR TITLE
feat: refine openmldb batch log and offline job info with SQL

### LINKED ISSUE #3632: Ignore the error message for offline jobs which try to register system tables without offline data
**Bug Description**

Now we may get the error message for all offline jobs which has created feature platform system tables.

```
2023-11-28 16:33:49,887 INFO api.OpenmldbSession: Register empty dataframe of SYSTEM_FEATURE_PLATFORM.feature_views with schema StructType(StructField(name,StringType,true), StructField(db,StringType,true), StructField(sql,StringType,true), StructField(description,StringType,true), StructField(feature_names,StringType,true))
2023-11-28 16:33:49,900 WARN api.OpenmldbSession: Fail to register table SYSTEM_FEATURE_PLATFORM.feature_views java.lang.NullPointerException
	at org.apache.spark.sql.Dataset.createOrReplaceTempView(Dataset.scala:3307)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.registerTable(OpenmldbSession.scala:238)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.$anonfun$registerOpenmldbOfflineTable$2(OpenmldbSession.scala:322)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.$anonfun$registerOpenmldbOfflineTable$2$adapted(OpenmldbSession.scala:284)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:198)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.$anonfun$registerOpenmldbOfflineTable$1(OpenmldbSession.scala:284)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:198)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.registerOpenmldbOfflineTable(OpenmldbSession.scala:282)
	at com._4paradigm.openmldb.batch.api.OpenmldbSession.<init>(OpenmldbSession.scala:76)
	at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:179)
	at org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:987)
	at com._4paradigm.openmldb.batchjob.RunBatchAndShow$.runBatchSql(RunBatchAndShow.scala:31)
	at com._4paradigm.openmldb.batchjob.RunBatchAndShow$.main(RunBatchAndShow.scala:27)
	at com._4paradigm.openmldb.batchjob.RunBatchAndShow.main(RunBatchAndShow.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:955)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1043)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1052)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

### LINKED ISSUE #3633: TaskManager job info show the temp path of sql file but not the sql string
**Describe the feature you'd like**

image

### PR DESCRIPTION
**Pull request by tobegit3hub**
Resolve https://github.com/4paradigm/OpenMLDB/issues/3633 .
Resolve https://github.com/4paradigm/OpenMLDB/issues/3632 .

### CODE REVIEW

**Comment by tobegit3hub** [comment]
Truncat the sql in `show jobs` command.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/api/OpenmldbSession.scala [hunk 1]
```diff
@@ -323,7 +325,7 @@ class OpenmldbSession {
             }
           } catch {
             case e: Exception => {
-              logger.warn(s"Fail to register table $dbName.$tableName " + ExceptionUtils.getStackTrace(e))
+              logger.warn(s"Fail to register table $dbName.$tableName, exception: " + e.getMessage)
```
Comment (vagetablechicken) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/api/OpenmldbSession.scala]: why revert printing stack? Some messages in our code is hard to understand

Comment (tobegit3hub) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/api/OpenmldbSession.scala]: Sometimes we may not register the online table without offline table info, the trace messages may be misunderstanding. We may handle this NPE first.

Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/api/OpenmldbSession.scala]: But when we need to debug why table register failed, the message is not enough. Maybe you can set the trace to debug level. If user meets the register error, user can set log level to get the trace.

Comment (tobegit3hub) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/api/OpenmldbSession.scala]: Okay, I will keep the code of print stacktrace and do not throw exception when registering system tables.

**Review by vagetablechicken — APPROVED** [review]

**Review by dl239 — APPROVED** [review]

File: java/pom.xml [hunk 2]
```diff
@@ -106,6 +106,20 @@
             </resource>
         </resources>
         <plugins>
+            <plugin>
```
Comment (aceforeverd) [inline java/pom.xml]: redefinition line 146

Comment (aceforeverd) [reply java/pom.xml]: handled already in https://github.com/4paradigm/OpenMLDB/blob/7edf33c5d6fad8883563d8b7d9f68d560fdceb89/java/pom.xml#L369-L380

Comment (tobegit3hub) [reply java/pom.xml]: done

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**